tests/core/reproducibility/test.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -eu -o pipefail
+
+function run_bazel() {
+    local output_base
+    local bazel_opts
+    local build_opts
+
+    output_base=$(mktemp -d)
+    bazel_opts=(
+    "--bazelrc=/dev/null"  # Disable any user specified remote/disk caching.
+    "--output_base=${output_base}"  # Use a disposable output_base.
+    )
+    build_opts=()
+
+    if [[ $(uname -s) == "Linux" ]]; then
+      # Use shared memory sandbox for faster runs.
+      build_opts+=("--experimental_sandbox_base=/dev/shm")
+    fi
+
+    bazel "${bazel_opts[@]}" build "${build_opts[@]+"${build_opts[@]}"}" \
```

Comment:
I'm not familiar with the syntax for expanding `build_opts` here. Can it just be `"${build_opts[@]}"`?

Reply:
Now removed, so no longer applicable. But the main reason was that empty arrays are considered unset in most bash versions. This will trigger the trap in `set -u` and the script fails. The syntax I had is the alternative value mechanism.
http://wiki.bash-hackers.org/syntax/pe#use_an_alternate_value